core.js

```diff
@@ -0,0 +1,685 @@
+  if (adminUnique){
+    //send current state to this admin
+    admin.sendAdd(this.getClients(), this.getRoutes(), this.getConnections());
+
+    //add the admin to the list of registered admins
+    this.admins.push(admin);
+  }
+  return adminUnique;
+};
+
+Manager.prototype.connectRoutePubClient = function(route, pubClient){
+  var publisher,
+      publisherI,
+      subClient,
+      subClientI,
+      subscriber,
+      subscriberI;
+
+  //if this client matches the route's "from" client
+  if(route.matchesFromClient(pubClient)){
+    //go through each publisher on this client
+    for(publisherI = pubClient.publishers.length - 1; 
+        publisherI >= 0; 
+        publisherI--){
+      publisher = pubClient.publishers[publisherI]; 
+      //if this publisher matches the route's "from" endpoint
+      if(route.matchesPublisher(pubClient, publisher)){
+        //then associate this publisher with this route
+        publisher.routes.push(route);
+        route.from.matched.push({client:pubClient,
+                                 endpoint:publisher});
+        //now because we want to handle pattern-matching with
+        // back-referencing from subscribers 
+        // to captured patterns in publishers
+        // we have to go through all subscribers in-context.
+
+        //so for each registered client
+        for(subClientI = this.clients.length - 1; 
+            subClientI >= 0;
+            subClientI--){
+          subClient = this.clients[subClientI];
+          //if this client matches the route's "to" client
+          if(route.matchesPubToClient(pubClient, publisher, subClient)){
+            //for each subscriber in that client
+            for(subscriberI = subClient.subscribers.length - 1;
+                subscriberI >= 0;
+                subscriberI--){
+              subscriber = subClient.subscribers[subscriberI];
+              //see if the subscriber matches this route/publisher combo
+              if(route.matchesPair(pubClient, publisher, 
+                                   subClient, subscriber)){
+                //and connect the subscriber to this publisher
+                pubClient.addOutConnection(publisher, 
+                                           subClient, 
+                                           subscriber, 
+                                           route);
+                //and connect the publisher to this subscriber
+                subClient.addInConnection(pubClient,
+                                          publisher,
+                                          subscriber,
+                                          route);
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+};
+
+/**
+ * Goes through the current list of routes and connects 
+ *   this client's endpoints to other endpoints based on 
+ *   the currently defined routes.
+ * @param {Leaf} client the client to connect
+ */
+Manager.prototype.connectClient = function(client){
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Function 'anonymous' has a complexity of 6.](https://www.codacy.com/app/quinkennedy/spacebrew/file/2279533613/issues/source?bid=3184377&fileBranchId=3187330#l140)